### ISSUE TITLE
[Paper Note] OPUS: Towards Efficient and Principled Data Selection in Large Language Model Pre-training in Every Iteration, Shaobo Wang+, arXiv'26, 2026.02

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#4516]
# URL
- https://arxiv.org/abs/2602.05400
# Authors
  - Shaobo Wang
  - Xuan Ouyang
  - Tianyi Xu
  - Yuzheng Hu
  - Jialin Liu
  - Guo Chen
  - Tianyu Zhang
  - Junhao Zheng
  - Kexin Yang
  - Xingzhang Ren
  - Dayiheng Liu
  - Linfeng Zhang
# Abstract
  - As high-quality public text approaches exhaustion, a phenomenon known as the Data Wall, pre-training is shifting from more tokens to better tokens. However, existing methods either rely on heuristic static filters that ignore training dynamics, or use dynamic yet optimizer-agnostic criteria based on raw gradients. We propose OPUS (Optimizer-induced Projected Utility Selection), a dynamic data selection framework that defines utility in the optimizer-induced update space. OPUS scores candidates by projecting their effective updates, shaped by modern optimizers, onto a target direction derived from a stable, in-distribution proxy. To ensure scalability, we employ Ghost technique with CountSketch for computational efficiency, and Boltzmann sampling for data diversity, incurring only 4.7\% additional compute overhead. OPUS achieves remarkable results across diverse corpora, quality tiers, optimizers, and model scales. In pre-training of GPT-2 Large/XL on FineWeb and FineWeb-Edu with 30B tokens, OPUS outperforms industrial-level baselines and even full 200B-token training. Moreover, when combined with industrial-level static filters, OPUS further improves pre-training efficiency, even with lower-quality data. Furthermore, in continued pre-training of Qwen3-8B-Base on SciencePedia, OPUS achieves superior performance using only 0.5B tokens compared to full training with 3B tokens, demonstrating significant data efficiency gains in specialized domains.
# Translation (by gpt-4o-mini)
- 高品質な公的テキストが枯渇に近づく中、「データウォール」と呼ばれる現象が現れ、事前トレーニングはより多くのトークンからより良いトークンへシフトしています。しかし、既存の手法は、トレーニングの動的特性を無視したヒューリスティックな静的フィルターに依存するか、生の勾配に基づいた動的ではあるが最適化器に依存しない基準を使用しています。そこで、我々はOPUS（最適化器誘導投影ユーティリティ選択）を提案します。これは、最適化器による更新空間でのユーティリティを定義する動的データ選択フレームワークです。OPUSは、現代の最適化器によって形成された効果的な更新を安定したイン分布プロキシから導き出されるターゲット方向に投影することによって候補をスコアリングします。スケーラビリティを確保するために、計算効率のためにCountSketchを使ったゴースト手法を採用し、データの多様性のためにボルツマンサンプリングを実施し、追加の計算オーバーヘッドはわずか4.7％に抑えています。OPUSは、多様なコーパス、品質レベル、最適化器、およびモデルのスケールで素晴らしい結果を達成しています。GPT-2 Large/XLをFineWebおよびFineWeb-Eduで30Bトークンを使った事前トレーニングにおいて、OPUSは産業レベルのベースラインを上回り、フル200Bトークンのトレーニングにも優れた成果を示しています。さらに、産業レベルの静的フィルターと組み合わせることで、OPUSは低品質データであっても事前トレーニングの効率をさらに向上させます。さらに、Qwen3-8B-BaseのSciencePediaにおける継続的な事前トレーニングにおいて、OPUSは3Bトークンでのフルトレーニングと比較して0.5Bトークンのみで優れたパフォーマンスを達成し、専門分野におけるデータ効率の大幅な向上を示しています。
# Summary (by gpt-4o-mini)
- 高品質な公的テキストが不足する中、データ選択の動的特性を無視した手法の限界を克服するために、最適化器誘導投影ユーティリティ選択（OPUS）を提案。OPUSは、効果的な更新を安定したプロキシから導き出すことでデータをスコアリングし、計算効率を考慮したゴースト手法とボルツマン・サンプリングを用いる。これにより、GPT-2 Large/XLやQwen3-8B-Baseにおいて優れた成果を上げ、事前トレーニングの効率を飛躍的に改善。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/_akhaliq/status/2021597620578185536?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
事前学習においてステップ単位で動的にバッチに含める学習データを選択する手法で、従来手法は単に勾配を考慮して選択していたが、実際にoptimizerによって更新される方向はmomentumなどによって異なるためgapが生じていた。これを埋めるために、optimizerが実際に重みを更新した際に、Validation setのlossがどれだけ低下するかによってUtilityを定義し、Utilityが大きくなるようにデータを動的に選択することで学習効率が向上する、といった話に見える。

<img width="838" height="388" alt="Image" src="https://github.com/user-attachments/assets/e45de40f-9fd3-4dcb-8e39-344de2fc9c57" />

<img width="756" height="408" alt="Image" src="https://github.com/user-attachments/assets/b7702daf-9540-4ff8-82f0-9a4bdc1ba42f" />

**Comment by AkihikoWatanabe** [comment]
著者ポスト:https://x.com/shaobowang6/status/2061672897845051695?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q
